Framework/PythonInterface/core/src/Converters/WrapWithNDArray.cpp

```diff
@@ -17,6 +17,20 @@
 
 #include <string>
 
+namespace {
+
+/**
+  Destructor for a capsule object being passed to Python. This helps release
+  memory of arrays passed to Python.
+  See: https://stackoverflow.com/questions/52731884/pyarray-simplenewfromdata
+*/
+void capsule_cleanup(PyObject *capsule) {
+  void *memory = PyCapsule_GetPointer(capsule, NULL);
+  delete[] memory;
```

Comment:
This might help you write a "clean" cleanup method :)
https://stackoverflow.com/questions/39305286/destroying-a-pycapsule-object?rq=1